packages/framework/src/hooks/useRegisterBindings.ts

```diff
@@ -40,12 +40,20 @@ export const useRegisterBindings = <T extends { [key: string]: unknown }>(
     isString(classStyles) ? classStyles : "",
   );
 
-  const styles = merge({}, styleProperties, values.styles);
-  if (!isEmpty(styles)) {
-    merge(values, { styles });
-  }
+  const styles = useMemo(() => {
+    return merge(
+      {},
+      styleProperties,
+      omit(values.styles ?? {}, ["className", "names"]),
+    );
```

Comment:
Instead of creating another hook here - let's combine this logic inside `useStyleNames` and rename the hook to `useStyles`